R/zzz.R

```diff
@@ -11,8 +11,24 @@
     assign(".ADDUCTS_MULT", add_multi, envir = asNamespace(pkgname))
     assign(".ADDUCTS_ADD", add_add, envir = asNamespace(pkgname))
     
-    substs_hmdb <- utils::read.table(system.file("substitutions/hmdb.txt",
-                                            package = "MetaboCoreUtils"),
-                                sep = "\t", header = TRUE)
-    assign(".SUBSTS_HMDB", substs_hmdb, envir = asNamespace(pkgname))
+    substs_hmdb_neutral <-
```

Comment:
I would suggest to maybe rename the txt files to `hmdb_neutral.txt`, `hmdb_positive.txt` etc and then load all files in the `substitutions` folder (in a loop) and assign each table with `assign(paste0(".", toupper(sub(".txt$", "", basename(fl)))...`

That way we could later add other definitions without the need to change the R code here.

Reply:
Or maybe even better, call the files "substs_hmdb_neutral.txt" - then you just need to add a `"."` before, make upper case and remove the file ending.

Reply:
would the code below be correct? Or is there another way to get files in a package besides `list.files`? I get an  error when running `rcmdcheck` ("object '.SUBSTS_HMDB_NEUTRAL' not found") and I wonder if it's wrong to use list.files. Again, beside `rcmdcheck` it works
```suggestion
       txts <- list.files("inst/substitutions", "*.txt", full.names = TRUE)
    for (txt in txts) {
        substs_hmdb <- utils::read.table(txt, sep = "\t", header = TRUE)
        assign(paste0(".", toupper(sub(".txt", "", basename(txt)))),
               substs_hmdb, envir = asNamespace(pkgname))
    }
```

Reply:
I think specifying the path like that `"inst/substitutions"` will not work. You'll need to use:

```r
txts <- dir(system.file("inst", "substitutions", package = "MetaboCoreUtils"), full.names = TRUE, pattern = "txt$")
```

then it *should* (?) hopefully work...

Reply:
Thanks, I think I've discovered the problem. By removing `"inst" ` from the code you suggested it works. I don't know why but that folder does not exist where the package is installed (although it exists in the repo).